Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp

```diff
@@ -263,6 +264,12 @@ template<typename Op> Calculation::Child toCalculationValue(const IndirectNode<O
     return Calculation::makeChild(WTF::apply([&](const auto& ...x) { return CalculationOp { toCalculationValue(x, options)... }; } , *root));
 }
 
+Calculation::Child toCalculationValue(const IndirectNode<Anchor>&, const ToConversionOptions&)
+{
+    ASSERT_NOT_REACHED("Unevaluated anchor() functions are not supported in the Calculation::Tree");
```

Comment:
Inelegant, but I don’t have a better idea.

Reply:
This is what we do with symbols and non-canonical dimension values, so it's in good company.

Reply:
That said, it kinda seems like we will need some for representation in the Calculation Tree, as it seems like the grammar allows for percentages in side the anchor function.

For instance:

```css
width: calc(10% + anchor(--x calc(20% / 2));
```  

Does the 20% inside the anchor get resolved by the style builder? Or does it have to wait until there is a container to do the math like other percentages?

Reply:
The percentage in the anchor function refers to the anchor side (left=0%, right=100%, percent values are between) and can be resolved at computed value time. You can have anything in the fallback but we handle that by simply replacing the anchor node with its fallback node during simplification so the anchor node does not show up in the calculation tree in that case either.